service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -134,14 +139,26 @@ protected void setCollaborators(
             adapter.setCommandConsumerFactory(commandConsumerFactory(adapterProperties, samplerFactory, commandRouterClient));
         }
 
+        final KafkaProducerConfigProperties kafkaProducerConfig = kafkaProducerConfig();
+        if (kafkaProducerConfig.getProducerConfig().isEmpty()) {
```

Comment:
IMHO we should better add a `KafkaProducerConfigProperties.isConfigured()` method in order to decouple the test from the inner structure of the properties object. WDYT?

Reply:
This is a good idea. I will do it.